### DISCUSSION TITLE
Pico_w LED remains off

### DISCUSSION THREAD

**Discussion by fatdollar** [mcknly/breadboard-os#11] [Q&A] [answered]
Yesterday I pulled down all the code and was watching your tutorial video trying to follow along.  I have an issue though were even when trying to turn on the LED it will always say LED is OFF.  Did I build the project incorrectly?

![Screenshot 2024-06-06 095045](https://github.com/mcknly/breadboard-os/assets/3020061/c6c2d190-3dac-4d6c-981b-00b4c983db9d)
I've attached a screen shot of the terminal and here is the `project.cmake` file:

```
 # PROJECT NAME - in quotes, no spaces
 set(PROJ_NAME "bbos-dev")
 
 # PROJECT VERSION - in quotes, no spaces, can contain alphanumeric if necessary
 set(PROJ_VER "0.1")
 
 # CLI INTERFACE - 0: use UART for CLI (default), 1: use USB for CLI
 set(CLI_IFACE 1)
 
 # MCU PLATFORM - set the MCU platform being used (i.e. the subdir in 'hardware/')
 set(PLATFORM rp2040)
 
 # BOARD - set the board being used (i.e. 'pico', 'pico_w', etc)
 set(BOARD pico_w)
```

I only worry about this because I wonder if the project is setup correctly to work with the other hardware on the board as well.

Any help would be appreciated. Thanks.
(Excellent project btw)

**Comment by mcknly** [comment, marked as answer]
It appears you may have found a bug. The onboard LED should not show up in the `dev/` menu when using Pico W, as the default GPIO used by the LED on non-W Pico is used by the wireless module. There is a way to toggle the onboard LED by sending a command to the wireless module, but that has not been implemented yet. Anyways the fact that it is returning OFF is actually correct because you are not actually able to toggle that GPIO. I am converting this conversation into an issue, thanks for reporting.

**Comment by fatdollar** [comment]
Well its always nice to know I'm not going insane. Thanks for the clarification.  So just since we're on the topic does that mean wireless support is also not implemented yet?

**Reply by mcknly** [reply]
Correct, nothing in the platform leverages the wireless module at this time. PR #9 will conditionally include the CYW43 driver library if the board is set to `pico_w`, which would be the first step in adding some Wi-Fi/BLE functionality to the project. If you have any ideas on what a basic feature set would look like, or would like to contribute, please open some discussions!

**Reply by fatdollar** [reply]
Awesome I'd love to contribute.  I may work on this LED bug first but I'll take a look at the PR and see where I can help.  Thanks for the awesome project and the great responsiveness.

**Reply by kintar** [reply]
@fatdollar : Take a look at discussion #19 and issue #20 .  I'm currently working on basic WiFi support, and the 0.2 release of BBOS has integrated the `pico_cyw43_arch_none` library into builds that are marked for pico_w. It doesn't yet initialize the cyw43 chip, however, and that's one of the things I'm working on integrating in the WiFi issue.

**Reply by fatdollar** [reply]
Thanks @Kintar if I can help out let me know but I'll look into those too.

**Reply by kintar** [reply]
@fatdollar : Actually, I just opened PR #21 for this. If you want to clone the branch and test it out, that would be helpful!
